companion-api/backend/users-file/users-file.go

```diff
@@ -49,18 +49,19 @@ func (fc *fileClient) CreateUser(id string, user *backend.UserData) (err error)
 
 }
 
-func (fc *fileClient) UpdateUser(id string, userToUpdate *backend.UserData, update func(old, new *backend.UserData)) (err error) {
-
-	var oldUser backend.UserData
-	if oldUser, err = fc.getUser(id); err == api.ErrMissingContent {
+func (fc *fileClient) UpdateUser(id string, patchs []backend.UserDataPatch) (err error) {
+	var user backend.UserData
+	user, err = fc.getUser(id)
+	if err != nil {
 		return
 	}
 
-	if update != nil {
-		update(&oldUser, userToUpdate)
+	err = user.ApplyPatchs(patchs)
```

Comment:
and, how incredible, logic here is replicated for each backend while with my proposed signature it wouldn't be ;-) and the user would have to know how to patch itself, too.